### PR TITLE
Check for DOM in CSSPluginBase.prototype.instantiate()

### LINKED ISSUE #111: CSSPluginBase.prototype.instantiate does not check if it is running in a browser.
See: https://github.com/systemjs/plugin-less/issues/6

When I try to use plugin-less in conjunction with server-side rendering, I get the following error:

```
err  (SystemJS) document is not defined
  ReferenceError: document is not defined
      at SystemJSNodeLoader.CSSPluginBase.instantiate (/path/to/jspm_packages/github/systemjs/plugin-css@0.1.30/css-plugin-base.js:64:24)
  Error loading /path/to/style.less!/path/to/jspm_packages/github/systemjs/plugin-less@0.1.2/less.js as "./style.less!" from /path/to/style.js
```

The root cause of the problem is `CSSPluginBase.prototype.instantiate()`, which attempts to inject a style element into the DOM without `document` being defined:

```javascript
CSSPluginBase.prototype.instantiate = function(load) {
  if (this.builder)
    return;

  var style = document.createElement('style');
  style.type = 'text/css';
  style.innerHTML = load.metadata.style;
  document.head.appendChild(style);
};
```

Explicitly checking if `document` is defined in `CSSPluginBase.prototype.instantiate()` seems appropriate and works for me in practice:

```javascript
  if (this.builder || typeof document === 'undefined')
    return;
```

But maybe that is not desirable for some reason?

### PR DESCRIPTION
**Pull request by mikol-styra**
Fixes #111

### CODE REVIEW

**Comment by guybedford** [comment]
Let me know if you have any ideas moving forward here. Otherwise I'll get to it when I get to it.

File: test/bundle.js [hunk 1]
```diff
@@ -4,6 +4,4 @@ System.registerDynamic("test/data/test.css!css.js", [], false, function ($__requ
   (function ($__global) {})(this);
 
   return _retrieveGlobal();
-});
-(function(c){if (typeof document == 'undefined') return; var d=document,a='appendChild',i='styleSheet',s=d.createElement('style');s.type='text/css';d.getElementsByTagName('head')[0][a](s);s[a](d.createTextNode(c));})
-("@import \"./dep.css\";body{background-color:red;background-image:url(test/data/x.png)}\n/*# sourceMappingURL=__.css.map */");
```
Comment (mikol-styra) [inline test/bundle.js]: Hrm… I am monkey patching CSSPluginBase in my environment and it builds just fine. I wonder what the difference is?

Comment (mikol-styra) [inline test/bundle.js]: Could be that plugin-less currently depends on plugin-css 0.1.30 and plugin-css itself is at 0.1.31.

Maybe the fix needs to be at the plugin-less level.

**Comment by mikol-styra** [comment]
Wait. One thing that might be happening is that the test is generating a bundle.css file… Not sure why that isn’t checked in, etc. But the CSS is being built.

**Comment by mikol-styra** [comment]
So I think that this change is _not_ breaking. The following test overwrites `test/bundle.js`, which is what I included in this PR.

```javascript
it('Should support separateCSS: true and sourceMaps: false', function() {
```

If I skip this test (with `xit`), then I think `bundle.js` comes out as expected. Namely:

```javascript
System.registerDynamic("test/data/test.css!css.js", [], false, function ($__require, $__exports, $__module) {
  var _retrieveGlobal = System.get("@@global-helpers").prepareGlobal($__module.id, null, null);

  (function ($__global) {})(this);

  return _retrieveGlobal();
});
(function(c){if (typeof document == 'undefined') return; var d=document,a='appendChild',i='styleSheet',s=d.createElement('style');s.type='text/css';d.getElementsByTagName('head')[0][a](s);s[a](d.createTextNode(c));})
("@import \"./dep.css\";body{background-color:red;background-image:url(test/data/x.png)}\n/*# sourceMappingURL=__.css.map */");
```

I’m not sure why `bundle.js` is checked in to begin with. But perhaps the tests should be updated to generate a separate bundle rather than overwriting the one file.

Let me know how you want to proceed, and I will update my PR accordingly.

Thanks!

**Comment by guybedford** [comment]
Ahh thanks, sorry yes I hadn't updated those files here.

Ok, looks great to me! Thanks again.

**Pull request merged**